modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -424,6 +432,27 @@ class Session implements ISession {
         return this
     }
 
+    /**
+     * Downloads the custom fsync file from the specified URL.
+     */
+    void downloadCustomFsync() {
+        Path customFsyncPath = workDir.resolve("custom_fsync")
+        if (customFsyncPath.exists()) {
+            log.debug "Custom fsync was already downloaded"
+            return
+        }
+
+        try {
+            URLConnection conn = new URL(CUSTOM_FSYNC_URL).openConnection()
+            conn.connect()
+            Files.copy(conn.inputStream, customFsyncPath)
+            customFsyncPath.setExecutable(true, false)
+            log.info "File downloaded successfully: $customFsyncPath"
```

Comment:
probably make this a debug log so customers don't get confused when  they dont recognize the file